dotnet/src/Extensions/PromptTemplates.Handlebars/Helpers/KernelHelpers/KernelFunctionHelpers.cs

```diff
@@ -22,18 +23,22 @@ internal static class KernelFunctionHelpers
     /// <param name="handlebarsInstance">The <see cref="IHandlebars"/>-context.</param>
     /// <param name="kernel">Kernel instance.</param>
     /// <param name="executionContext">Kernel arguments maintained as the executing context.</param>
+    /// <param name="promptConfig">The associated prompt template configuration.</param>
+    /// <param name="allowUnsafeContent">Flag indicating whether to allow unsafe content</param>
     /// <param name="nameDelimiter">The character used to delimit the plugin name and function name in a Handlebars template.</param>
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
     public static void Register(
         IHandlebars handlebarsInstance,
         Kernel kernel,
         KernelArguments executionContext,
+        PromptTemplateConfig promptConfig,
+        bool allowUnsafeContent,
         string nameDelimiter,
         CancellationToken cancellationToken)
     {
         foreach (var function in kernel.Plugins.GetFunctionsMetadata())
         {
-            RegisterFunctionAsHelper(kernel, executionContext, handlebarsInstance, function, nameDelimiter, cancellationToken);
+            RegisterFunctionAsHelper(kernel, executionContext, handlebarsInstance, function, allowUnsafeContent || promptConfig.AllowUnsafeContent, nameDelimiter, cancellationToken);
```

Comment:
Why do we have two ways of setting this? More maintenance, is there a use-case I'm not aware, maybe moving the `bool` to be an extension where it sets the `AllowUnsafeContent` in the `PromptTemplateConfig` ?

Reply:
If you set `AllowUnsafeContent` for the factory it will trust all content which essentially reverts the behaviour back to how things currently work. The reason for doing this is to provide a way for applications that rely on the current behaviour to keep working. Although we will encourage developers to migrate to the new safer approach.